### DISCUSSION TITLE
Is SSO authentication there? or can we set up the SSO instead of LDAP authentication in codechecker?

### DISCUSSION THREAD

**Discussion by gaurav00009** [Ericsson/codechecker#3928] [General]
Hi,

I am using codechecker for the last 1 year. Everything is good here except authentication because most of the time the user doesn't want to enter his credentials. So if SSO is there then it will help everyone and it is also a secure method of authentication. I was looking for SSO articles but I didn't find anything related to SSO. Can you please suggest if is there any article or how can we set up SSO for authentication?

**Comment by bruntib** [comment]
Hi,

CodeChecker doesn't have SSO authentication. This document describes the authentication methods: https://github.com/Ericsson/codechecker/blob/master/docs/web/authentication.md. Consider using personal authentication tokens that can prevent entering credentials all the time.

**Reply by gaurav00009** [reply]
Hi,

Thanks for the information @bruntib.
Are there any plans for SSO in a later version? Please let me know as we are waiting for the version which will have SSO authorization.

**Comment by dkrupp** [comment]
Hi @gaurav00009, 
what type of SSO would work best for you? Could you please elaborate a bit you reuqirements?
thanks.

**Reply by gaurav00009** [reply]
Hi @dkrupp 

Basically we need SAML Single Sign-On (SSO) for Azure AD users. Because most of the applications we use are SAML based, so it would be a good move if SAML would be implemented here.
For example
Like if I have logged in with my Outlook account then in that case codechecker should also log in with SAML. I hope you understand my point that we need.

Thanks,
Gaurav
